code/sound/audiostr.cpp

```diff
@@ -922,6 +923,47 @@ void audiostream_close()
 
 }
 
+static int audiostream_use_next_free( int type )
+{
+    if ( !Audiostream_inited || !snd_is_inited() )
+        return -1;
+
+    int i;
+    for (i = 0; i < MAX_AUDIO_STREAMS; i++)
+        if (Audio_streams[i].status == ASF_FREE)
+            break;
+
+    if (i == MAX_AUDIO_STREAMS) {
+        nprintf(("Sound", "SOUND => No more audio streams available!\n"));
+        return -1;
+    }
+
+    Audio_streams[i].status = ASF_USED;
+    Audio_streams[i].type = type;
+
+    switch (type) {
+        case ASF_SOUNDFX: // As in: sound.cpp:590
+            Audio_streams[i].Set_Default_Volume(Master_sound_volume * aav_effect_volume);
+            break;
+        case ASF_EVENTMUSIC: // As in: sexp.cpp:11562
+            Audio_streams[i].Set_Default_Volume(Master_event_music_volume * aav_music_volume);
+            break;
+        case ASF_MENUMUSIC: // As in: mainhallmenu.cpp:1170
+            Audio_streams[i].Set_Default_Volume(Master_event_music_volume);
+            break;
+        case ASF_VOICE: // As in: sound.cpp:590
+            Audio_streams[i].Set_Default_Volume(Master_voice_volume * aav_voice_volume);
+            break;
+        default:
+            Audio_streams[i].status = ASF_FREE;
+            return -1;
+    }
+
+    Audio_streams[i].Set_Default_Volume(type == ASF_VOICE ? Master_voice_volume * aav_voice_volume : Master_sound_volume);
```

Comment:
This will overwrite the value set previously. Is that intended?

Reply:
That was an oversight, left in from before I realized there are more than the two defaults exposed via LUA. Correcting.